src/pages/index.js

```diff
@@ -7,43 +7,61 @@ import "./index.css";
  *=============================================**/
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
-import { initialCards, config } from "../utils/constants.js";
+import {
+  config,
+  apiConfig,
+  openProfileButton,
+  openAvatarButton,
+  cardSelector,
+  cardContainer,
+  openCardButton,
+  cardElement,
+} from "../utils/constants.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupDeleteConfirmation from "../components/PopupDeleteConfirmation.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
-/**============================================
- * Profile and profile modal selector variables
- *=============================================**/
-const openProfileButton = document.querySelector(".profile__edit-button");
-/**============================================
- *       Card template selector variables
- *=============================================**/
-const cardSelector = "#card-template";
-const cardContainer = document.querySelector(".card__container");
-const openCardButton = document.querySelector(".profile__add-button");
+import Api from "../components/Api.js";
 /**============================================
  *            Class Instantiations
  *=============================================**/
 const userInfo = new UserInfo({
   name: ".profile__name",
   description: ".profile__description",
+  avatar: ".profile__image",
 });
-
+const api = new Api(apiConfig);
 const profilePopupForm = new PopupWithForm(
   "#modal__profile",
   handleProfileSubmit
 );
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  cardContainer
-);
+const avatarPopupForm = new PopupWithForm("#modal__avatar", handleAvatarSubmit);
+const cardSection = new Section({}, cardContainer);
 const cardPopupForm = new PopupWithForm("#modal__add-card", handleCardSubmit);
 const popupWithImage = new PopupWithImage("#modal__preview-wrapper");
+const deleteConfirmation = new PopupDeleteConfirmation(
+  "#modal__delete-card",
+  deleteConfirmed
+);
+const card = new Card(
+  cardElement,
+  cardSelector,
+  handleImageClick,
+  handleDeleteClick,
+  handleLikeClick
+);
 /**============================================
  *      Validation and Card Class Calls
  *=============================================**/
-cardSection.renderItems();
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then((res) => {
+    const userData = res[0];
+    const cards = Array.from(res[1]);
```

Comment:
**Could be improved**


You can destructure the response

```js

Promise.all([api.getUserInfo(), api.getInitialCards()])
  .then(([userData, cards]) => {
    getInitialApiUserInfo(userData);
    handleApiCardInfo(cards);
  })
  .catch(console.error);
```

Reply:
Great point, this cleaned up the code quite a bit.